internal/component/prometheus/exporter/static/static.go

```diff
@@ -0,0 +1,108 @@
+package static
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/expfmt"
+
+	"github.com/grafana/alloy/internal/component"
+	"github.com/grafana/alloy/internal/component/prometheus/exporter"
+	"github.com/grafana/alloy/internal/featuregate"
+	"github.com/grafana/alloy/internal/static/integrations"
+	"github.com/grafana/alloy/internal/static/integrations/config"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:      "prometheus.exporter.static",
+		Stability: featuregate.StabilityExperimental,
+		Args:      Arguments{},
+		Exports:   exporter.Exports{},
+
+		Build: exporter.New(createExporter, "static"),
+	})
+}
+
+func createExporter(opts component.Options, args component.Arguments, defaultInstanceKey string) (integrations.Integration, string, error) {
+	a := args.(Arguments)
+	return integrations.NewIntegrationWithInstanceKey(opts.Logger, a.Convert(), opts.ID)
+}
+
+type Arguments struct {
+	Text string `alloy:"text,attr"`
```

Comment:
I have not implement the full proposal made in https://github.com/grafana/alloy/issues/1637 as we discussed in slack a bit. But I do wonder if we want to make this an array instead so you would specify something like:
```
prometheus.exporter.static "demo" {
  text {
    value = ""
  }
}
```

The reason for that would be if we want to extend this to include alloy syntax for metrics, like the proposal have, it would play better.

A block with the same structure would also be sufficient because that I am pretty sure it would make it backward compatible to convert it to an array in the future.